analysis/type.go

```diff
@@ -99,6 +100,56 @@ func (a *DefaultAnalyzer) Analyze(input []byte) TokenStream {
 
 var ErrInvalidDateTime = fmt.Errorf("unable to parse datetime with any of the layouts")
 
+const UnixSecs = "unix_sec"
+const UnixMilliSecs = "unix_milli"
+const UnixMicroSecs = "unix_micro"
+const UnixNanoSecs = "unix_nano"
+
+type TimestampBounds struct {
+	Min int64
+	Max int64
+}
+
+var UnixTimestampFormats = map[string]*TimestampBounds{
+	UnixSecs: {
+		Min: math.MinInt64 / 1000000000,
+		Max: math.MaxInt64 / 1000000000,
+	},
+	UnixMilliSecs: {
+		Min: math.MinInt64 / 1000000,
+		Max: math.MaxInt64 / 1000000,
+	},
+	UnixMicroSecs: {
+		Min: math.MinInt64 / 1000,
+		Max: math.MaxInt64 / 1000,
+	},
+	UnixNanoSecs: {
+		Min: math.MinInt64,
+		Max: math.MaxInt64,
+	},
+}
+
+func convertTimestamp(timestamp int64, format string) int64 {
+	if format == UnixSecs {
+		return timestamp * 1000000000
+	} else if format == UnixMilliSecs {
+		return timestamp * 1000000
+	} else if format == UnixMicroSecs {
+		return timestamp * 1000
+	}
+	return timestamp
+}
+
+// ValidateAndConvertTimestamp validates the timestamp against the bounds and
+// converts it to the nanoseconds if valid.
+func ValidateAndConvertTimestamp(timestamp int64, bounds *TimestampBounds, format string) (int64, error) {
+	if timestamp > bounds.Min && timestamp < bounds.Max {
+		return convertTimestamp(timestamp, format), nil
+	} else {
```

Comment:
this else condition is not needed, might as well return the 0, error at the end of the function. it looks more readable that way i think